packages/headless/src/controllers/product-listing/headless-product-listing.ts

```diff
@@ -0,0 +1,61 @@
+import {executeProductListingSearch} from '../../features/product-listing/product-listing-actions';
+import {buildController} from '../controller/headless-controller';
+import {Schema, SchemaValues} from '@coveo/bueno';
+import {configuration, productListing} from '../../app/reducers';
+import {loadReducerError} from '../../utils/errors';
+import {ProductListingEngine} from '../../app/product-listing-engine/product-listing-engine';
+
+// TODO COM-1185 - Actually add options and dispatch the updates
+const optionsSchema = new Schema({});
+
+export type ProductListingListOptions = SchemaValues<typeof optionsSchema>;
+
+export interface ProductListingListProps {
+  options?: ProductListingListOptions;
+}
+
+export type ProductListingList = ReturnType<typeof buildProductListing>;
+export type ProductListingListState = ProductListingList['state'];
+
+export const buildProductListing = (
```

Comment:
I'd suggest `buildProductList` for the name, since the controller seems like the analogue of the `ResultList` for a search page?